packages/mobile/src/send/utils.ts

```diff
@@ -105,3 +118,49 @@ export function showLimitReachedError(
 
   return showError(ErrorMessages.PAYMENT_LIMIT_REACHED, ALERT_BANNER_DURATION, translationParams)
 }
+
+export function* handleSendPaymentData(data: UriData, cachedRecipient?: RecipientWithContact) {
+  const recipient: RecipientWithQrCode = {
+    kind: RecipientKind.QrCode,
+    address: data.address,
+    displayId: data.e164PhoneNumber,
+    displayName: data.displayName || cachedRecipient?.displayName || 'anonymous',
+    phoneNumberLabel: cachedRecipient?.phoneNumberLabel,
+    thumbnailPath: cachedRecipient?.thumbnailPath,
+    contactId: cachedRecipient?.contactId,
+  }
+  yield put(storeLatestInRecents(recipient))
+
+  if (data.currencyCode) {
+    yield put(selectPreferredCurrency(data.currencyCode as LocalCurrencyCode))
```

Comment:
Can UriData be typed correctly to prevent the need for casting here?

Reply:
enums in typescript are weird
it's typed as the value strings of the enum which aren't recognized by typescript as the enum type